cosmic_ray/plugins.py

```diff
@@ -1,24 +1,48 @@
 """Query and retrieve the various plugins in Cosmic Ray.
 """
 
+import logging
+
 from stevedore import driver, ExtensionManager
 
+log = logging.getLogger()
+
+
+def _log_extension_loading_failure(_mgr, ep, err):
+    log.info('Operator provider load failure: extension-point="%s", err="%s"', ep, err)
+
+
+OPERATOR_PROVIDERS = {
+    extension.name: extension.plugin()
+    for extension in
+    ExtensionManager(
+        'cosmic_ray.operator_providers',
+        on_load_failure_callback=_log_extension_loading_failure)
+}
+
 
 def get_operator(name):
-    """Get an operator class from a plugin.
+    """Get an operator class from a provider plugin.
 
     Attrs:
-        name: The name of the plugin containing the operator class.
+        name: The name of the operator class.
 
-    Returns: The operator *class object* (i.e. not an instance) provided by the
-        plugin named `name`.
+    Returns: The operator *class object* (i.e. not an instance).
     """
-    return ExtensionManager('cosmic_ray.operators')[name].plugin
+    sep = name.index('/')
```

Comment:
Can you use:

    provider_name, _, operator_name = name.partition('/')

here?